Make Climesync PEP8 Compliant

Now that @aaroncohen73 has made this a respectable project, let's make it a respectable project. 

TODO
----

* Add flake8 to `requirements.txt`
* Fix `climesync.py` until `flake8 climesync.py` no longer returns errors.

Output of `flake8 climesync.py` on a recent PR:

```
climesync.py:6:1: F401 'json' imported but unused
climesync.py:39:10: E261 at least two spaces before inline comment
climesync.py:41:1: E302 expected 2 blank lines, found 1
climesync.py:42:80: E501 line too long (83 > 79 characters)
climesync.py:63:1: E302 expected 2 blank lines, found 1
climesync.py:65:1: W293 blank line contains whitespace
climesync.py:68:80: E501 line too long (83 > 79 characters)
climesync.py:69:1: W293 blank line contains whitespace
climesync.py:73:80: E501 line too long (91 > 79 characters)
climesync.py:105:1: W293 blank line contains whitespace
climesync.py:115:80: E501 line too long (83 > 79 characters)
climesync.py:121:1: E302 expected 2 blank lines, found 1
climesync.py:122:80: E501 line too long (87 > 79 characters)
climesync.py:126:80: E501 line too long (82 > 79 characters)
climesync.py:131:80: E501 line too long (81 > 79 characters)
climesync.py:137:9: F841 local variable 'optional_prompt' is assigned to but never used
climesync.py:139:9: F841 local variable 'type_prompt' is assigned to but never used
climesync.py:143:29: E261 at least two spaces before inline comment
climesync.py:147:29: E261 at least two spaces before inline comment
climesync.py:151:29: E261 at least two spaces before inline comment
climesync.py:166:1: E302 expected 2 blank lines, found 1
climesync.py:178:18: W291 trailing whitespace
climesync.py:180:1: E302 expected 2 blank lines, found 1
climesync.py:190:1: E302 expected 2 blank lines, found 1
climesync.py:191:80: E501 line too long (90 > 79 characters)
climesync.py:205:1: E302 expected 2 blank lines, found 1
climesync.py:206:80: E501 line too long (106 > 79 characters)
climesync.py:212:1: W293 blank line contains whitespace
climesync.py:220:18: W291 trailing whitespace
climesync.py:222:1: E302 expected 2 blank lines, found 1
climesync.py:231:74: E502 the backslash is redundant between brackets
climesync.py:232:74: E502 the backslash is redundant between brackets
climesync.py:233:74: E502 the backslash is redundant between brackets
climesync.py:234:74: E502 the backslash is redundant between brackets
climesync.py:235:74: E502 the backslash is redundant between brackets
climesync.py:244:1: E302 expected 2 blank lines, found 1
climesync.py:255:68: E502 the backslash is redundant between brackets
climesync.py:256:59: E502 the backslash is redundant between brackets
climesync.py:257:52: E502 the backslash is redundant between brackets
climesync.py:258:65: E502 the backslash is redundant between brackets
climesync.py:259:75: E502 the backslash is redundant between brackets
climesync.py:260:58: E502 the backslash is redundant between brackets
climesync.py:262:1: W293 blank line contains whitespace
climesync.py:266:1: E302 expected 2 blank lines, found 1
climesync.py:276:63: E502 the backslash is redundant between brackets
climesync.py:277:69: E502 the backslash is redundant between brackets
climesync.py:278:72: E502 the backslash is redundant between brackets
climesync.py:279:62: E502 the backslash is redundant between brackets
climesync.py:280:57: E502 the backslash is redundant between brackets
climesync.py:281:80: E501 line too long (80 > 79 characters)
climesync.py:281:80: E502 the backslash is redundant between brackets
climesync.py:282:78: E502 the backslash is redundant between brackets
climesync.py:285:80: E501 line too long (93 > 79 characters)
climesync.py:288:1: E302 expected 2 blank lines, found 1
climesync.py:289:80: E501 line too long (104 > 79 characters)
climesync.py:291:56: E502 the backslash is redundant between brackets
climesync.py:292:64: E502 the backslash is redundant between brackets
climesync.py:317:1: E302 expected 2 blank lines, found 1
climesync.py:326:80: E501 line too long (87 > 79 characters)
climesync.py:334:22: W291 trailing whitespace
climesync.py:336:1: E302 expected 2 blank lines, found 1
climesync.py:345:55: E502 the backslash is redundant between brackets
climesync.py:346:58: E502 the backslash is redundant between brackets
climesync.py:347:54: E502 the backslash is redundant between brackets
climesync.py:353:1: E302 expected 2 blank lines, found 1
climesync.py:364:64: E502 the backslash is redundant between brackets
climesync.py:365:67: E502 the backslash is redundant between brackets
climesync.py:366:62: E502 the backslash is redundant between brackets
climesync.py:372:1: E302 expected 2 blank lines, found 1
climesync.py:382:80: E501 line too long (84 > 79 characters)
climesync.py:382:84: E502 the backslash is redundant between brackets
climesync.py:383:80: E501 line too long (81 > 79 characters)
climesync.py:383:81: E502 the backslash is redundant between brackets
climesync.py:386:80: E501 line too long (83 > 79 characters)
climesync.py:389:1: E302 expected 2 blank lines, found 1
climesync.py:398:80: E501 line too long (90 > 79 characters)
climesync.py:399:1: W293 blank line contains whitespace
climesync.py:406:22: W291 trailing whitespace
climesync.py:408:1: E302 expected 2 blank lines, found 1
climesync.py:417:56: E502 the backslash is redundant between brackets
climesync.py:423:1: E302 expected 2 blank lines, found 1
climesync.py:434:65: E502 the backslash is redundant between brackets
climesync.py:440:1: E302 expected 2 blank lines, found 1
climesync.py:450:80: E501 line too long (85 > 79 characters)
climesync.py:450:85: E502 the backslash is redundant between brackets
climesync.py:451:80: E501 line too long (83 > 79 characters)
climesync.py:451:83: E502 the backslash is redundant between brackets
climesync.py:454:80: E501 line too long (83 > 79 characters)
climesync.py:457:1: E302 expected 2 blank lines, found 1
climesync.py:466:80: E501 line too long (90 > 79 characters)
climesync.py:476:1: E302 expected 2 blank lines, found 1
climesync.py:485:64: E502 the backslash is redundant between brackets
climesync.py:486:64: E502 the backslash is redundant between brackets
climesync.py:487:73: E502 the backslash is redundant between brackets
climesync.py:488:59: E502 the backslash is redundant between brackets
climesync.py:489:80: E501 line too long (80 > 79 characters)
climesync.py:489:80: E502 the backslash is redundant between brackets
climesync.py:490:80: E501 line too long (84 > 79 characters)
climesync.py:490:84: E502 the backslash is redundant between brackets
climesync.py:491:80: E501 line too long (88 > 79 characters)
climesync.py:491:88: E502 the backslash is redundant between brackets
climesync.py:492:65: E502 the backslash is redundant between brackets
climesync.py:498:1: E302 expected 2 blank lines, found 1
climesync.py:509:64: E502 the backslash is redundant between brackets
climesync.py:510:64: E502 the backslash is redundant between brackets
climesync.py:511:72: E502 the backslash is redundant between brackets
climesync.py:512:58: E502 the backslash is redundant between brackets
climesync.py:513:76: E502 the backslash is redundant between brackets
climesync.py:514:80: E501 line too long (80 > 79 characters)
climesync.py:514:80: E502 the backslash is redundant between brackets
climesync.py:515:80: E501 line too long (84 > 79 characters)
climesync.py:515:84: E502 the backslash is redundant between brackets
climesync.py:516:63: E502 the backslash is redundant between brackets
climesync.py:522:1: E302 expected 2 blank lines, found 1
climesync.py:534:80: E501 line too long (83 > 79 characters)
climesync.py:537:1: E302 expected 2 blank lines, found 1
climesync.py:546:80: E501 line too long (90 > 79 characters)
climesync.py:554:22: E261 at least two spaces before inline comment
climesync.py:556:1: E302 expected 2 blank lines, found 1
climesync.py:557:80: E501 line too long (82 > 79 characters)
climesync.py:562:22: E261 at least two spaces before inline comment
climesync.py:565:8: E111 indentation is not a multiple of four
climesync.py:636:1: E302 expected 2 blank lines, found 1
climesync.py:640:80: E501 line too long (80 > 79 characters)
climesync.py:658:1: W293 blank line contains whitespace
```

--- a/climesync.py
+++ b/climesync.py
@@ -3,7 +3,6 @@
 import pymesync
 import sys
 import argparse
-import json
 
 menu_options = (
     "===============================================================\n"
@@ -36,10 +35,11 @@ menu_options = (
 arg_username = ""
 arg_password = ""
 timesync_url = ""
-ts = None # pymesync.TimeSync object
+ts = None  # pymesync.TimeSync object
+
 
 def print_json(response):
-    """Prints a list of dictionaries (such as those returned by pymesync) nicely"""
+    """Prints values returned by Pymesync nicely"""
 
     print ""
 
@@ -62,17 +62,18 @@ def print_json(response):
 
     print ""
 
+
 def get_field(prompt, optional=False, field_type=""):
-    """Prompts the user for input and returns it in an optionally specified format
-    
+    """Prompts the user for input and returns it in the specified format
+
     prompt - The prompt to display to the user
     optional - Whether or not the field is optional (defaults to False)
-    field_type - The type of input. If left empty, it returns the input as a string
-    
+    field_type - The type of input. If left empty, input is a string
+
     Valid field_types:
     ? - Yes/No input
     # - Integer input
-    ! - Multiple inputs delimited by commas, stripped of whitespace, and returned as a list
+    ! - Multiple inputs delimited by commas returned as a list
     """
 
     # If necessary, add extra prompts that inform the user
@@ -105,7 +106,7 @@ def get_field(prompt, optional=False, field_type=""):
             if field_type == "?":
                 if response.upper() in ["Y", "YES", "N", "NO"]:
                     return True if response.upper() in ["Y", "YES"] else False
-            
+
             elif field_type == "#":
                 if response.isdigit():
                     return int(response)
@@ -116,43 +117,42 @@ def get_field(prompt, optional=False, field_type=""):
             elif field_type == "":
                 return response
 
-            # If the provided field_type doesn't make sense, return an empty string
+            # If the provided field_type isn't valid, return an empty string
             else:
                 return ""
 
         print "Please submit a valid input"
 
+
 def get_fields(fields):
-    """Prompts the user for multiple input fields and returns the responses in a dictionary
+    """Prompts for multiple fields and returns everything in a dictionary
 
     fields - A list of tuples that are ordered (field_name, prompt)
 
-    field_name can contain special characters that signify specific types of input
+    field_name can contain special characters that signify input type
     ? - Yes/No field
     # - Integer field
     ! - List field
 
-    In addition to those, field_name can contain a * to signify an optional field
+    In addition to those, field_name can contain a * for an optional field
     """
     responses = dict()
 
     for field, prompt in fields:
         optional = False
-        optional_prompt = ""
         field_type = ""
-        type_prompt = ""
 
         # Deduce field type
         if "?" in field:
-            field_type = "?" # Yes/No question
+            field_type = "?"  # Yes/No question
             field = field.replace("?", "")
 
         elif "#" in field:
-            field_type = "#" # Integer
+            field_type = "#"  # Integer
             field = field.replace("#", "")
 
         elif "!" in field:
-            field_type = "!" # Comma-delimited list
+            field_type = "!"  # Comma-delimited list
             field = field.replace("!", "")
 
         if "*" in field:
@@ -167,17 +167,21 @@ def get_fields(fields):
 
     return responses
 
+
 def get_user_permissions(users):
-    """Asks for permissions for a list of users and returns them in a sane format for pymesync"""
+    """Asks for permissions for multiple users and returns them in a dict"""
 
     permissions = dict()
 
     for user in users:
         user_permissions = dict()
 
-        member = get_field("Is %s a project member?" % user, field_type="?")
-        spectator = get_field("Is %s a project spectator?" % user, field_type="?")
-        manager = get_field("Is %s a project manager?" % user, field_type="?")
+        member = get_field("Is {} a project member?".format(user),
+                           field_type="?")
+        spectator = get_field("Is {} a project spectator?".format(user),
+                              field_type="?")
+        manager = get_field("Is {} a project manager?".format(user),
+                            field_type="?")
 
         user_permissions["member"] = member
         user_permissions["spectator"] = spectator
@@ -186,6 +190,7 @@ def get_user_permissions(users):
         permissions[user] = user_permissions
 
     return permissions
+
 
 def connect():
     """Creates a new pymesync.TimeSync instance with a new URL"""
@@ -199,7 +204,8 @@ def connect():
     ts = pymesync.TimeSync(baseurl=timesync_url)
 
     # No response from server upon connection
-    return list() 
+    return list()
+
 
 def disconnect():
     """Disconnects from the TimeSync server"""
@@ -211,8 +217,9 @@ def disconnect():
     # No response from server
     return list()
 
+
 def sign_in():
-    """Attempts to sign in with credentials provided on the command line or by the user"""
+    """Attempts to sign in with user-supplied or command line credentials"""
 
     global arg_username, arg_password, ts
 
@@ -226,14 +233,15 @@ def sign_in():
     # Attempt to authenticate and return the server's response
     return ts.authenticate(username, password, "password")
 
+
 def sign_out():
-    """Signs out from TimeSync and resets the command line credentials so the user can input them again"""
+    """Signs out from TimeSync and resets command line credentials"""
 
     global arg_username, arg_password, timesync_url, ts
 
     if not ts:
         return {"error": "Not connected to TimeSync server"}
-    
+
     # Reset the credentials provided on the command line
     arg_username = arg_password = ""
 
@@ -241,7 +249,8 @@ def sign_out():
     ts = pymesync.TimeSync(baseurl=timesync_url)
 
     # No response from server
-    return list() 
+    return list()
+
 
 def create_time():
     """Creates a new time and submits it to the TimeSync server"""
@@ -252,18 +261,19 @@ def create_time():
         return {"error": "Not connected to TimeSync server"}
 
     # The data to send to the server containing the new time information
-    post_data = get_fields([("#duration", "Duration in seconds"),        \
-                            ("project", "Project slug"),                 \
-                            ("!activities", "Activity slugs"),           \
-                            ("date_worked", "Date worked (yyyy-mm-dd)"), \
-                            ("*issue_uri", "Issue URI"),                 \
-                            ("*notes", "Notes")])
+    post_data = get_fields([("#duration",   "Duration in seconds"),
+                            ("project",     "Project slug"),
+                            ("!activities", "Activity slugs"),
+                            ("date_worked", "Date worked (yyyy-mm-dd)"),
+                            ("*issue_uri",  "Issue URI"),
+                            ("*notes",      "Notes")])
 
     # Use the currently authenticated user
     post_data["user"] = ts.user
 
     # Attempt to create a time and return the response
     return ts.create_time(time=post_data)
+
 
 def update_time():
     """Sends revised time information to the TimeSync server"""
@@ -276,16 +286,17 @@ def update_time():
     uuid = get_field("UUID of time to update")
 
     # The data to send to the server containing revised time information
-    post_data = get_fields([("*#duration", "Duration in seconds"), \
-                            ("*project", "Project slug"), \
-                            ("*user", "New user"), \
-                            ("*!activities", "Activity slugs"), \
-                            ("*date_worked", "Date worked (yyyy-mm-dd)"), \
-                            ("*issue_url", "Issue URI"), \
-                            ("*notes", "Notes")])
-    
+    post_data = get_fields([("*#duration",   "Duration in seconds"),
+                            ("*project",     "Project slug"),
+                            ("*user",        "New user"),
+                            ("*!activities", "Activity slugs"),
+                            ("*date_worked", "Date worked (yyyy-mm-dd)"),
+                            ("*issue_url",   "Issue URI"),
+                            ("*notes",       "Notes")])
+
     # Attempt to update a time and return the reponse
     return ts.update_time(uuid=uuid, time=post_data)
+
 
 def get_times():
     """Queries the TimeSync server for submitted times with optional filters"""
@@ -297,23 +308,24 @@ def get_times():
 
     # Optional filtering parameters to send to the server
     print "Filtering times..."
-    post_data = get_fields([("*!user", "Submitted by users"), \
-                            ("*!project", "Belonging to projects"), \
-                            ("*!activity", "Belonging to activities"), \
-                            ("*start", "Beginning on date"), \
-                            ("*end", "Ending on date"), \
-                            ("*?include_revisions", "Include revised times?"), \
-                            ("*?include_deleted", "Include deleted times?"), \
+    post_data = get_fields([("*!user", "Submitted by users"),
+                            ("*!project", "Belonging to projects"),
+                            ("*!activity", "Belonging to activities"),
+                            ("*start", "Beginning on date"),
+                            ("*end", "Ending on date"),
+                            ("*?include_revisions", "Include revised times?"),
+                            ("*?include_deleted", "Include deleted times?"),
                             ("*uuid", "By UUID")])
 
-    # Attempt to query the server for times with filtering parameters and return the response
+    # Attempt to query the server for times with filtering parameters
     return ts.get_times(query_parameters=post_data)
 
-def sum_times():
-    """Sums all the times associated with a specific project and between optional start and end dates"""
 
-    query = get_fields([("!project", "Project slugs"), \
-                        ("*start", "Start date (yyyy-mm-dd)"), \
+def sum_times():
+    """Sums all the times associated with a specific project"""
+
+    query = get_fields([("!project", "Project slugs"),
+                        ("*start", "Start date (yyyy-mm-dd)"),
                         ("*end", "End date (yyyy-mm-dd)")])
 
     result = ts.get_times(query)
@@ -338,6 +350,7 @@ def sum_times():
         print e
         return result
 
+
 def delete_time():
     """Deletes a time from the TimeSync server"""
 
@@ -347,7 +360,8 @@ def delete_time():
         return {"error": "Not connected to TimeSync server"}
 
     uuid = get_field("Time UUID")
-    really = get_field("Do you really want to delete time {}?".format(uuid), field_type="?")
+    really = get_field("Do you really want to delete {}?".format(uuid),
+                       field_type="?")
 
     # If the user really wants to delete it
     if really:
@@ -355,7 +369,8 @@ def delete_time():
 
     # If no, return an empty list
     else:
-        return list() 
+        return list()
+
 
 def create_project():
     """Creates a new project on the TimeSync server"""
@@ -366,10 +381,10 @@ def create_project():
         return {"error": "Not connected to TimeSync server"}
 
     # The data to send to the server containing new project information
-    post_data = get_fields([("name", "Project name"), \
-                            ("!slugs", "Project slugs"), \
-                            ("*uri", "Project URI"), \
-                            ("*!users", "Users"), \
+    post_data = get_fields([("name", "Project name"),
+                            ("!slugs", "Project slugs"),
+                            ("*uri", "Project URI"),
+                            ("*!users", "Users"),
                             ("*default_activity", "Default activity")])
 
     # If users have been added to the project, ask for user permissions
@@ -379,6 +394,7 @@ def create_project():
 
     # Attempt to create a new project and return the response
     return ts.create_project(project=post_data)
+
 
 def update_project():
     """Sends revised project information to the TimeSync server"""
@@ -391,10 +407,10 @@ def update_project():
     slug = get_field("Slug of project to update")
 
     # The data to send to the server containing revised project information
-    post_data = get_fields([("*name", "Updated project name"), \
-                            ("*!slugs", "Updated project slugs"), \
-                            ("*uri", "Updated project URI"), \
-                            ("*!users", "Updated users"), \
+    post_data = get_fields([("*name", "Updated project name"),
+                            ("*!slugs", "Updated project slugs"),
+                            ("*uri", "Updated project URI"),
+                            ("*!users", "Updated users"),
                             ("*default_activity", "Updated default activity")])
 
     # If user permissions are going to be updated, ask for them
@@ -404,6 +420,7 @@ def update_project():
 
     # Attempt to update the project information and return the response
     return ts.update_project(project=post_data, slug=slug)
+
 
 def get_projects():
     """Queries the TimeSync server for projects with optional filters"""
@@ -415,12 +432,13 @@ def get_projects():
 
     # Optional filtering parameters
     print "Filtering projects..."
-    post_data = get_fields([("*?include_revisions", "Include project revisions?"), \
-                            ("*?include_deleted", "Include deleted projects?"), \
+    post_data = get_fields([("*?include_revisions", "Include revisions?"),
+                            ("*?include_deleted", "Include deleted?"),
                             ("*!slugs", "By project slug")])
 
-    # Attempt to query the server with filtering parameters and return the response
+    # Attempt to query the server with filtering parameters
     return ts.get_projects(query_parameters=post_data)
+
 
 def delete_project():
     """Deletes a project from the TimeSync server"""
@@ -431,15 +449,17 @@ def delete_project():
         return {"error": "Not connected to TimeSync server"}
 
     slug = get_field("Project slug")
-    really = get_field("Do you really want to delete project {}?".format(slug), field_type="?")
-    
+    really = get_field("Do you really want to delete {}?".format(slug),
+                       field_type="?")
+
     # If the user really wants to delete it
     if really:
         return ts.delete_project(slug=slug)
 
     # If no, return an empty list
     else:
-        return list() 
+        return list()
+
 
 def create_activity():
     """Creates a new activity on the TimeSync server"""
@@ -450,11 +470,12 @@ def create_activity():
         return {"error": "Not connected to TimeSync server"}
 
     # The data to send to the server containing new activity information
-    post_data = get_fields([("name", "Activity name"), \
+    post_data = get_fields([("name", "Activity name"),
                             ("slug", "Activity slug")])
 
     # Attempt to create a new activity and return the response
     return ts.create_activity(activity=post_data)
+
 
 def update_activity():
     """Sends revised activity information to the TimeSync server"""
@@ -467,11 +488,12 @@ def update_activity():
     slug_to_update = get_field("Slug of activity to update")
 
     # The data to send to the server containing revised activity information
-    post_data = get_fields([("*name", "Updated activity name"), \
+    post_data = get_fields([("*name", "Updated activity name"),
                             ("*slug", "Updated activity slug")])
 
     # Attempt to update the activity information and return the repsonse
     return ts.update_activity(activity=post_data, slug=slug_to_update)
+
 
 def get_activities():
     """Queries the TimeSync server for activities with optional filters"""
@@ -483,12 +505,13 @@ def get_activities():
 
     # Optional filtering parameters
     print "Filtering activities..."
-    post_data = get_fields([("*?include_revisions", "Include activity revisions?"), \
-                            ("*?include_deleted", "Include deleted activities?"), \
+    post_data = get_fields([("*?include_revisions", "Include revisions?"),
+                            ("*?include_deleted", "Include deleted?"),
                             ("*slug", "By activity slug")])
 
-    # Attempt to query the server with filtering parameters and return the response
+    # Attempt to query the server with filtering parameters
     return ts.get_activities(query_parameters=post_data)
+
 
 def delete_activity():
     """Deletes an activity from the TimeSync server"""
@@ -499,7 +522,8 @@ def delete_activity():
         return {"error": "Not connected to TimeSync server"}
 
     slug = get_field("Activity slug")
-    really = get_field("Do you really want to delete activity {}?".format(slug), field_type="?")
+    really = get_field("Do you really want to delete {}?".format(slug),
+                       field_type="?")
 
     # If the user really wants to delete it
     if really:
@@ -508,6 +532,7 @@ def delete_activity():
     # If no, return an empty list
     else:
         return list()
+
 
 def create_user():
     """Creates a new user on the TimeSync server"""
@@ -518,18 +543,19 @@ def create_user():
         return {"error": "Not connected to TimeSync server"}
 
     # The data to send to the server containing new user information
-    post_data = get_fields([("username", "New user username"), \
-                            ("password", "New user password"), \
-                            ("*display_name", "New user display name"), \
-                            ("*email", "New user email"), \
-                            ("*?site_admin", "Is the new user a site admin?"), \
-                            ("*?site_manager", "Is the new user a site manager?"), \
-                            ("*?site_spectator", "Is the new user a site spectator?"), \
-                            ("*meta", "Extra metainformation"), \
+    post_data = get_fields([("username", "New user username"),
+                            ("password", "New user password"),
+                            ("*display_name", "New user display name"),
+                            ("*email", "New user email"),
+                            ("*?site_admin", "Is the user an admin?"),
+                            ("*?site_manager", "Is the user a manager?"),
+                            ("*?site_spectator", "Is the user a spectator?"),
+                            ("*meta", "Extra meta-information"),
                             ("*?active", "Is the new user active?")])
 
     # Attempt to create a new user and return the response
     return ts.create_user(user=post_data)
+
 
 def update_user():
     """Sends revised user information to the TimeSync server"""
@@ -542,18 +568,19 @@ def update_user():
     username_to_update = get_field("Username of user to update")
 
     # The data to send to the server containing revised user information
-    post_data = get_fields([("*username", "Updated username"), \
-                            ("*password", "Updated password"), \
-                            ("*display_name", "Updated display name"), \
-                            ("*email", "Updated email"), \
-                            ("*?site_admin", "Is the user a site admin?"), \
-                            ("*?site_manager", "Is the user a site manager?"), \
-                            ("*?site_spectator", "Is the user a site spectator?"), \
-                            ("*meta", "New metainformation"), \
+    post_data = get_fields([("*username", "Updated username"),
+                            ("*password", "Updated password"),
+                            ("*display_name", "Updated display name"),
+                            ("*email", "Updated email"),
+                            ("*?site_admin", "Is the user an admin?"),
+                            ("*?site_manager", "Is the user a manager?"),
+                            ("*?site_spectator", "Is the user a spectator?"),
+                            ("*meta", "New metainformation"),
                             ("*?active", "Is the user active?")])
 
     # Attempt to update the user and return the response
     return ts.update_user(user=post_data, username=username_to_update)
+
 
 def get_users():
     """Queries the TimeSync server for users with optional filters"""
@@ -567,8 +594,9 @@ def get_users():
     print "Filtering users..."
     username = get_field("By username", optional=True)
 
-    # Attempt to query the server with filtering parameters and return the response
+    # Attempt to query the server with filtering parameters
     return ts.get_users(username=username)
+
 
 def delete_user():
     """Deletes a user from the TimeSync server"""
@@ -579,7 +607,8 @@ def delete_user():
         return {"error": "Not connected to TimeSync server"}
 
     username = get_field("Username")
-    really = get_field("Do you really want to delete user {}?".format(username), field_type="?")
+    really = get_field("Do you really want to delete {}?".format(username),
+                       field_type="?")
 
     # If the user really wants to delete it
     if really:
@@ -587,18 +616,19 @@ def delete_user():
 
     # If no, return an empty list
     else:
-        return list() # If no, return an empty list
+        return list()
+
 
 def menu():
-    """Provides the user with menu options and executes the commands they input"""
+    """Provides the user with menu options and executes commands"""
 
     print menu_options
 
     choice = raw_input(">> ")
-    response = list() # A list of python dictionaries
+    response = list()  # A list of python dictionaries
 
     if choice == "c":
-       response = connect()
+        response = connect()
 
     elif choice == "dc":
         response = disconnect()
@@ -669,11 +699,12 @@ def menu():
     # Print server response
     print_json(response)
 
+
 def main():
     global arg_username, arg_password, timesync_url, ts
 
     parser = argparse.ArgumentParser()
-    parser.add_argument("-c", "--connect", help="connect to a timesync baseurl")
+    parser.add_argument("-c", "--connect", help="connect to a timesync server")
     parser.add_argument("-u", "--username", help="specify your username")
     parser.add_argument("-p", "--password", help="specify your password")
 
@@ -691,7 +722,7 @@ def main():
 
     if args.password:
         arg_password = args.password
-    
+
     # If all args are provided, attempt to sign in
     if timesync_url and arg_username and arg_password:
         print_json(sign_in())

--- a/requirements.txt
+++ b/requirements.txt
@@ -1,1 +1,2 @@
 pymesync
+flake8

